Modules/_testcapimodule.c

```diff
@@ -1559,6 +1559,19 @@ getargs_et_hash(PyObject *self, PyObject *args)
     return result;
 }
 
+static PyObject *
+get_indices(PyObject *self, PyObject *args)
+{
+  PySliceObject *slice;
+  Py_ssize_t length, start, stop, step;
+
+  if (!PyArg_ParseTuple(args, "On", &slice, &length))
+    return NULL;
+
+  int result = PySlice_GetIndices(slice, length, &start, &stop, &step);
```

Comment:
Return NULL if an error is set.

Reply:
The method `PySlice_GetIndices` only returns an int and I couldn't see it raising any error. I might be missing something on returning NULL based on error here. Is it about returning NULL based on result to be -1? I am still a beginner in C trying to improve so please bear with me on this point.

Reply:
I think you meant the below code : 
```
if (result == -1 && PyErr_Occurred())
    return NULL
```

Reply:
Right. Actually `result == -1 &&` is not needed, it is used only for performance, but in this case it doesn't matter. So you can write
```c
if (PyErr_Occurred()) {
    assert(result == -1);
    return NULL;
}
```

`PyInt_AsSsize_t()` in `PySlice_GetIndices` can raise an OverflowError.